### PR TITLE
Added httpsender scripts

### PR DESCRIPTION
**Pull request by rezen**
@kingthorin hey, have had my resources tied up! Here is the first PR for my scripts.

### CODE REVIEW

**Comment by kingthorin** [comment]
@rezen any chance you could fix those issues that LGTM found?

Also there are maintain auth variants but not the matching auth scripts? (Or I'm misunderstanding something)

**Comment by cudandres** [comment]
Sorry, I saw all green I didn't pay attention to the alerts!

The *-maintain-auth scripts pair up with JSON/Form-based auth configs, no need for custom scripts.

**Comment by kingthorin** [comment]
Great thanks!

File: httpsender/inject-xss.js [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+    print(this['zap.script.name'] + '] ' +  arguments[0]);
+}
+
+
+var ignoreKeys = [
+   'id', 'status'
+];
+
+// @todo
+// var signatures = {};
```
Comment (kingthorin) [inline httpsender/inject-xss.js]: Remove this?

File: httpsender/inject-xss.js [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+
+function injectXssPayloads(data) {
+  if (Array.isArray(data)) {
+	  /*
+    for (var i in data) {
+      data[i] = injectXssPayloads(data[i]);
+    }
+    */
```
Comment (kingthorin) [inline httpsender/inject-xss.js]: Remove?

File: httpsender/juice-shop-maintain-auth.js [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+
+var HttpSender    = Java.type('org.parosproxy.paros.network.HttpSender');
+var ScriptVars    = Java.type('org.zaproxy.zap.extension.script.ScriptVars');
+
+function sendingRequest(msg, initiator, helper) {  
+  var headers = msg.getRequestHeader();
+  var url     = headers.getURI().toString();
+  // var qry     = headers.getURI().getQuery();
+  // var cookies = headers.getCookieParams();
+  // var reqbody = msg.getRequestBody().toString();
```
Comment (kingthorin) [inline httpsender/juice-shop-maintain-auth.js]: Remove these?

File: httpsender/keep-cookies-going.js [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+function logger() {
+  print('[' + this['zap.script.name'] + '] ' + arguments[0]);
+}
+
+var ScriptVars    = Java.type('org.zaproxy.zap.extension.script.ScriptVars');
+var HtmlParameter = Java.type('org.parosproxy.paros.network.HtmlParameter')
+var COOKIE_TYPE   = org.parosproxy.paros.network.HtmlParameter.Type.cookie;
+
+function sendingRequest(msg, initiator, helper) {  
+  // var reqbody = msg.getRequestBody().toString();
```
Comment (kingthorin) [inline httpsender/keep-cookies-going.js]: Remove?

File: httpsender/keep-cookies-going.js [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+  var headers = msg.getRequestHeader();
+  var cookies = headers.getCookieParams();
+
+  // var url     = headers.getURI().toString();
+  // var qry     = headers.getURI().getQuery();
```
Comment (kingthorin) [inline httpsender/keep-cookies-going.js]: Remove?

File: httpsender/keep-cookies-going.js [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+}
+
+// If a cookie was set, capture it
+function responseReceived(msg, initiator, helper) {
+  var resheaders = msg.getResponseHeader();
+  var setCookie  = resheaders.getHeader('Set-Cookie');
+  // var headers    = msg.getRequestHeader();
+  // var url        = headers.getURI().toString();
+  // var reqbody    = msg.getRequestBody().toString();
+  // var resbody    = msg.getResponseBody().toString();
```
Comment (kingthorin) [inline httpsender/keep-cookies-going.js]: Remove these?

File: httpsender/maintain-jwt.js [hunk 7]
```diff
@@ -0,0 +1,69 @@
+/*exported sendingRequest, responseReceived*/
+// Logging with the script name is super helpful!
+function logger() {
+  print('[' + this['zap.script.name'] + '] ' + arguments[0]);
+}
+
+// Control.getSingleton().getExtensionLoader().getExtension(ExtensionUserManagement.class);
```
Comment (kingthorin) [inline httpsender/maintain-jwt.js]: Remove?

File: httpsender/maintain-jwt.js [hunk 8]
```diff
@@ -0,0 +1,69 @@
...
+
+function sendingRequest(msg, initiator, helper) {  
+  if (initiator === HttpSender.AUTHENTICATION_INITIATOR) {
+    logger("Trying to auth")
+  return msg;
+  }
+
+  var token = ScriptVars.getGlobalVar("jwt-token")
+  if (!token) {return;}
+  // var headers = msg.getRequestHeader();
```
Comment (kingthorin) [inline httpsender/maintain-jwt.js]: Remove?

File: httpsender/spider-wrangler.js [hunk 9]
```diff
@@ -0,0 +1,119 @@
...
+  if (historyRef.getStatusCode() > 299) {
+    return false;
+  } 
+  var test_url = historyRef.getURI().toString();
+  test_url = stripUrlExtras(test_url);
+  return (url === test_url);   
+}
+
+// Check if url is in SiteNode history
+// http://www.zaproxy.org/2.5/javadocs/org/parosproxy/paros/model/SiteNode.html
```
Comment (kingthorin) [inline httpsender/spider-wrangler.js]: Update?
https://static.javadoc.io/org.zaproxy/zap/2.8.0/org/parosproxy/paros/model/SiteNode.html

**Comment by kingthorin** [comment]
All comments addressed, rebased current,

**Comment by thc202** [comment]
Thank you both!

**Pull request merged**